File: unit_testing_fundamentals/src/test/java/com/techyourchance/unittestingfundamentals/example1/PositiveNumberValidatorTest.java
package com.techyourchance.unittestingfundamentals.example1;

import static org.hamcrest.CoreMatchers.is;

import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;

public class PositiveNumberValidatorTest {

    PositiveNumberValidator SUT;

    @Before
    public void setup() {
        SUT = new PositiveNumberValidator();
    }

    @Test
    public void test1() {
        boolean result = SUT.isPositive(-1);
        Assert.assertThat(result, is(false));
    }

    @Test
    public void test2() {
        boolean result = SUT.isPositive(0);
        Assert.assertThat(result, is(false));
    }

    @Test
    public void test3() {
        boolean result = SUT.isPositive(1);
        Assert.assertThat(result, is(true));
    }
}

File: unit_testing_fundamentals/src/test/java/com/techyourchance/unittestingfundamentals/example1/PositiveNumberValidatorTest.kt
package com.techyourchance.unittestingfundamentals.example1

import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test

class PositiveNumberValidatorTest {

    private val validator: PositiveNumberValidator = PositiveNumberValidator()

    @Test
    fun test1() {
        assertFalse(validator.isPositive(-1))
        assertFalse(validator.isPositive(-10))
        assertFalse(validator.isPositive(-900))
        assertFalse(validator.isPositive(-50))
        assertFalse(validator.isPositive(-10000))
    }

    @Test
    fun test2() {
        assertFalse(validator.isPositive(0))
    }

    @Test
    fun test3() {
        assertTrue(validator.isPositive(1))
        assertTrue(validator.isPositive(10))
        assertTrue(validator.isPositive(900))
        assertTrue(validator.isPositive(50))
        assertTrue(validator.isPositive(10000))

    }
}